tools/spice/spice_utils.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Optional
+
+import spiceypy as spice
+from spiceypy.utils.exceptions import NotFoundError
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class SpiceKernelManager:
+    """Manages the loading and handling of SPICE kernels.
+
+    Parameters
+    ----------
+    kernel_path : Path
+        The local directory path where the SPICE kernel files are stored.
+    extensions : list of str, optional
+        A list of file extensions to be considered as valid SPICE kernel files.
```

Comment:
We could only allow the default ones. I think it's useful during development to be able to select a single extension that I want to furnish and then view the constants for only that kernel type. But this could totally be used without passing the extensions and just relying on the default ones.